src/app/components/audio-recordings/pages/download/download.component.ts

```diff
@@ -0,0 +1,389 @@
+    return {
+      key: "projectId",
+      type: "select",
+      defaultValue: this.project.id,
+      templateOptions: {
+        label: "Project",
+        // TODO Allow user to change selected site
+        disabled: true,
+        options: [{ value: this.project.id, label: this.project.name }],
+      },
+    };
+  }
+
+  private getRegionFormInput(hideProject: boolean): FormlyFieldConfig {
+    return {
+      key: "regionId",
+      type: "select",
+      defaultValue: this.region.id,
+      templateOptions: {
+        label: hideProject ? "Project" : "Site",
+        // TODO Allow user to change selected site
+        disabled: true,
+        options: [{ value: this.region.id, label: this.region.name }],
+      },
+    };
+  }
+
+  private getSiteFormInput(hideProject: boolean): FormlyFieldConfig {
+    return {
+      key: "siteId",
+      type: "select",
+      defaultValue: this.site.id,
+      templateOptions: {
+        label: hideProject ? "Site" : "Point",
+        // TODO Allow user to change selected site
+        disabled: true,
+        options: [{ value: this.site.id, label: this.site.name }],
+      },
+    };
+  }
+
+  private dateSelectionRange(): FormlyFieldConfig[] {
+    return [
+      {
+        key: "recordingStartedBefore",
+        type: "input",
+        templateOptions: {
+          label: "Date recording started before",
+          required: false,
+          type: "datetime-local",
+          description:
+            "The UTC (+00:00) time zone is used for all date filtering. " +
+            "If you're unsure if your data will be included in a filter, expand your date range by one-day in either direction.",
+        },
+      },
+      {
+        key: "recordingFinishedBefore",
+        type: "input",
+        templateOptions: {
+          label: "Date recording ended before",
+          required: false,
+          type: "datetime-local",
+          description:
+            "The UTC (+00:00) time zone is used for all date filtering. " +
+            "If you're unsure if your data will be included in a filter, expand your date range by one-day in either direction.",
+        },
+      },
+    ];
+  }
+
+  public updateHref(): void {
+    if (!this.project) {
+      return;
+    }
+    this.href = this.recordingsApi.batchDownloadUrl(this.generateFilter());
+  }
+
+  private generateFilter(): Filters<AudioRecording> {
+    const filter: Filters<AudioRecording> = {};
+    const recordingStartedBefore = DateTime.fromISO(
+      this.model.recordingStartedBefore
+    );
+    const recordingFinishedBefore = DateTime.fromISO(
+      this.model.recordingFinishedBefore
+    );
+
+    if (this.model.siteId) {
+      filter["sites.id"] = { equal: this.model.siteId };
+    } else if (this.model.regionId) {
+      filter["sites.regions.id"] = { equal: this.model.regionId };
+    } else if (this.model.projectId) {
+      filter["sites.regions.project.id"] = {
+        equal: this.model.projectId,
+      };
+    }
```

Comment:
feels dirty to have bits of filter query arbitrarily constructed here... remind me to get you to reinvent our QueryBuilder class

Reply:
https://github.com/QutEcoacoustics/workbench-client/issues/1777